File: test/llvm_swift_test_code/linear_constant/global_15.swift
var g1 = 0
var g2 = 99

struct X {
  init() {
    g1 = 1024
  }
}

class Y {
  init() {
    g2 = g2 + 1
  }
  deinit {
    g1 = g2 + 13
  }
}

var v = X()
var w = Y()
@main
struct MyMain {

  static func foo (_ x: Int)-> Int {
    var a = x
    a = a + 1
    return a
  }

  static func main() {
    var a = g1
    var b = g2
    a = foo(a)
    a = a + 30
  }
}

File: test/llvm_test_code/linear_constant/global_15.cpp
int g1 = 0;
int g2 = 99;

struct X {
  X() { g1 = 1024; }
};

struct Y {
  Y() { g2 = g2 + 1; }
  ~Y() { g1 = g2 + 13; }
};

int foo(int x) {
  x = x + 1;
  return x;
}

X var;
Y war;

int main() {
  int a = g1;
  int b = g2;
  a = foo(a);
  return a + 30;
}
